src/herder/PendingEnvelopes.cpp

```diff
@@ -179,6 +180,17 @@ PendingEnvelopes::isNodeInQuorum(NodeID const& node)
                         m->statement);
                     res = getQSet(h);
                 }
+                if (res == nullptr)
+                {
+                    // see if we had some information for that node
+                    auto& db = mApp.getDatabase();
```

Comment:
This seems to me like it risks doing a SQL query on every SCP message (including those from strangers), even when they're returning null. I think we probably need a little more caching of the results here, both positive and negative.

Reply:
no it won't:
rebuild is only happening when we update quorum information which only happens when we actually process an SCP message, and we only process SCP messages for nodes in the known transitive quorum.

Reply:
ok, so only known people -- but is it still potentially an awful lot of messages?